controllers/update.go

```diff
@@ -327,6 +327,10 @@ func (r *ComponentDetectionQueryReconciler) updateComponentStub(req ctrl.Request
 			},
 		}
 
+		if len(componentPortsMap[context]) != 0 {
+			componentStub.TargetPort = componentPortsMap[context][0]
```

Comment:
does the first value always correspond to the first kubernetes component if set?

Reply:
The ports here correspond to the ports returned by alizer, which were detected from the application source code - they should always be in order.